src/openklant/components/klantinteracties/api/validators.py

```diff
@@ -235,6 +233,33 @@ def __call__(self, value: str):
                 "No channels to validate `kanaal` were found for the configured tabel_code in the Referentielijsten API."
             )
 
+        now = datetime.now(timezone.utc)
+
+        def is_valid_by_geldigheid(item):
+            begin = item.get("beginGeldigheid")
+            eind = item.get("eindGeldigheid")
+
+            begin_dt = (
+                datetime.fromisoformat(begin.replace("Z", "+00:00")) if begin else None
+            )
+            eind_dt = (
+                datetime.fromisoformat(eind.replace("Z", "+00:00")) if eind else None
+            )
+
+            if begin_dt and now < begin_dt:
+                return False
+            if eind_dt and now >= eind_dt:
+                return False
+            return True
```

Comment:
This is more easy to understand i think, and is the same like here?
`Only consider items valid if the current timestamp >= beginGeldigheid and < eindGeldigheid`
```suggestion
            if begin_dt and eind_dt:
                return begin_dt <= now < eind_dt
            return True
```

however, i am not sure if the validation is still valid if the dates are None. ?!, i don't know.

Add test also for these cases